cmd/internal/cli/key_newpair.go

```diff
@@ -6,34 +6,194 @@
 package cli
 
 import (
+	"errors"
+	"fmt"
 	"os"
 
+	"github.com/sylabs/singularity/internal/pkg/util/interactive"
+
+	"github.com/sylabs/singularity/pkg/cmdline"
+	"github.com/sylabs/singularity/pkg/sypgp"
+
 	"github.com/spf13/cobra"
 	"github.com/sylabs/singularity/docs"
 	scs "github.com/sylabs/singularity/internal/pkg/remote"
 	"github.com/sylabs/singularity/internal/pkg/sylog"
-	"github.com/sylabs/singularity/pkg/sypgp"
 )
 
-// KeyNewPairCmd is 'singularity key newpair' and generate a new OpenPGP key pair
-var KeyNewPairCmd = &cobra.Command{
-	Args:                  cobra.ExactArgs(0),
-	DisableFlagsInUseLine: true,
-	PreRun:                sylabsToken,
-	Run: func(cmd *cobra.Command, args []string) {
-		keyring := sypgp.NewHandle("")
-		handleKeyNewPairEndpoint()
-
-		if _, err := keyring.GenKeyPair(keyServerURI, authToken, keyNewpairBitLength); err != nil {
-			sylog.Errorf("creating newpair failed: %v", err)
-			os.Exit(2)
+var (
+	keyNewPairName     string
+	KeyNewPairNameFlag = &cmdline.Flag{
+		ID:           "KeyNewPairNameFlag",
+		Value:        &keyNewPairName,
+		DefaultValue: "",
+		Name:         "name",
+		ShortHand:    "N",
+		Usage:        "keys owner name",
+	}
+
+	keyNewPairEmail     string
+	KeyNewPairEmailFlag = &cmdline.Flag{
+		ID:           "KeyNewPairEmailFlag",
+		Value:        &keyNewPairEmail,
+		DefaultValue: "",
+		Name:         "email",
+		ShortHand:    "E",
+		Usage:        "keys owner email",
+	}
+
+	keyNewPairComment     string
+	KeyNewPairCommentFlag = &cmdline.Flag{
+		ID:           "KeyNewPairCommentFlag",
+		Value:        &keyNewPairComment,
+		DefaultValue: "",
+		Name:         "comment",
+		ShortHand:    "C",
+		Usage:        "keys comment",
+	}
+
+	keyNewPairPassword     string
+	KeyNewPairPasswordFlag = &cmdline.Flag{
+		ID:           "KeyNewPairPasswordFlag",
+		Value:        &keyNewPairPassword,
+		DefaultValue: "",
+		Name:         "password",
+		ShortHand:    "P",
+		Usage:        "keys password",
+	}
+
+	keyNewPairPush     bool
+	KeyNewPairPushFlag = &cmdline.Flag{
+		ID:           "KeyNewPairPushFlag",
+		Value:        &keyNewPairPush,
+		DefaultValue: false,
+		Name:         "push",
+		ShortHand:    "U",
+		Usage:        "push generated key to a keystore",
```

Comment:
```suggestion
		Usage:        "specify to push the public key to the remote keystore (default true)",
```

And make the default behavior to push/true. Because that is the default when interactively generating a newpair.

Also suggest adding a example in `docs.KeyNewPairExample` of generating a newpair non-interactively, with `--push=false`.